registry/remote/example_test.go

```diff
@@ -0,0 +1,424 @@
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag) // First resolve the tag to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example blob content
+}
+
+// ExampleRepository_Fetch_byDigest gives example snippets for downloading a blob by digest.
+func ExampleRepository_Fetch_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // Still need to resolve first, don't create a new descriptor with the digest, blob size is unknown
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+
+	// option 1: fetch a WHOLE blob
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+
+	// option 2: partially read, if the remote registry supports
+	if seeker, ok := r.(io.ReadSeeker); ok {
+		_, err = seeker.Seek(8, io.SeekStart)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		pulledBlob, err := io.ReadAll(r)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		fmt.Println(string(pulledBlob))
+	}
+
+	// caller SHOULD verify what is fetched
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) || descriptor.Size != int64(len(pulledBlob)) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example blob content
+	// blob content
+}
+
+// ExampleRepository_Tag gives example snippets for tagging a descriptor.
+func ExampleRepository_Tag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// suppose we are going to tag a blob with below digest
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+
+	// 1. Resolve the target desc
```

Comment:
nit: The variable name is `descriptor`, but the comment calls it `desc`.
It's `descriptor` in other comments.